src/Http/Request.php

```diff
@@ -213,4 +220,33 @@ public static function getBatchApi()
     {
         return static::OUTLOOK_ROOT_URL . static::OUTLOOK_VERSION . DIRECTORY_SEPARATOR . self::OUTLOOK_BATCH_ENDPOINT;
     }
+
+    public function getNewHeader(string $url, array $params = []): array
+    {
+        $token = !empty($params['token']) ? $params['token'] : null;
+        if (!empty($token['clientID']) && !empty($token['clientSecret']) && !empty(OUTLOOK_PROXY_URL)) {
```

Comment:
please check if you can use arguments to pass this proxy url, also an asset check for all these is good enuf I think